File: src/main/java/iesvdm/adrian/proyecto/domain/Partido.java
package iesvdm.adrian.proyecto.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import org.springframework.data.util.Pair;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@EqualsAndHashCode(of = "id")
@Builder
@Table(name = "partido")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Partido implements Serializable{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_partido")
    private Long id;

    private Date fecha;

    private int resultadoLocal;

    private int resultadoVisitante;

    private String nombre_local;

    private String nombre_visitante;

    @JsonIgnore
    @ManyToOne
    private Torneo torneo;

}

File: crud-front/src/app/model/partido.ts
import { Equipo } from "./equipo";
import { Torneo } from "./torneo";

export class Partido {
    id!: number;
    local: string;
    visitante: string;
    resultadoLocal: number;
    resultadoVisitante: number;
    torneo: Torneo;
    
    constructor(local: string, visitante: string, torneo: Torneo, resultadoLocal: number,  resultadoVisitante: number){
        this.resultadoLocal = resultadoLocal;
        this.resultadoVisitante = resultadoVisitante;
        this.local = local;
        this.visitante = visitante;
        this.torneo = torneo;
    }
}
